--- origsrc/mod_clamav-0.22/sample.conf	2004-03-21 17:25:53.000000000 -0600
+++ src/mod_clamav-0.22/sample.conf	2008-05-16 04:48:31.203125000 -0500
@@ -8,13 +8,15 @@
 
 # during make install, includes the module in httpd.conf, so the fllowing
 # load directive is very seldom needed
-LoadModule clamav_module      modules/mod_clamav.so
+<IfModule !mod_clamav.c>
+LoadModule clamav_module      lib/apache2/modules/mod_clamav.so
+</IfModule>
 
 # specify the directory where the module should place files durin download
 ClamavTmpdir    /var/tmp/clamav
 
 # the db directory is only needed in local mode
-ClamavDbdir     /usr/local/share/clamav
+ClamavDbdir     /usr/share/clamav
 
 # bypass scanning of jpeg images
 ClamavSafetypes image/jpg
@@ -32,8 +34,8 @@
 # names for shared memory and mutex. Note that we don't know exactly what
 # apache does in the background. However, we should make sure that apache
 # can create these files if necessary
-ClamavShm	/usr/local/apache2/logs/clamav.shm
-ClamavMutex	/usr/local/apache2/logs/clamav.lock
+ClamavShm	/var/log/apache2/clamav.shm
+ClamavMutex	/var/log/apache2/clamav.lock
 
 # if the daemon crashes, we will have a problem connecting to it. Since
 # we don't have any PCs, we are not very paranoid about this
